CodeU_Assignement_1/src/codeu_assignement_1/main.java

```diff
@@ -0,0 +1,278 @@
+        // Gets the k-th location out-of the k-th to the last location
+        int k = l.size() - k2last - 1;
+        if (k < 0 || k > l.size()) 
+            throw new IndexOutOfBoundsException();
+    
+        return l.get(k);
+    } // End Q2
+    
+    // Some testings to the algorithms
+    static void run_tests_Q1()
+    {
+       String s1 = null;
+       String s2 = null;
+       System.out.println("Q1: (true) " + s1 + (q1_permutation(s1,s2) ? " is " : " isn't ") 
+                + "a permutation of " + s2);
+       
+       s1 = "";
+       System.out.println("Q1: (false) " + s1 + (q1_permutation(s1,s2) ? " is " : " isn't ") 
+                + "a permutation of " + s2);
+       
+       s2 = "";
+       System.out.println("Q1: (true) " + s1 + (q1_permutation(s1,s2) ? " is " : " isn't ") 
+                + "a permutation of " + s2);
+       
+       s2 = "a";
+       System.out.println("Q1: (false) " + s1 + (q1_permutation(s1,s2) ? " is " : " isn't ") 
+                + "a permutation of " + s2);
+       
+       s1 = null;
+       System.out.println("Q1: (false) " + s1 + (q1_permutation(s1,s2) ? " is " : " isn't ") 
+                + "a permutation of " + s2);
+       
+       s1 = "";
+       System.out.println("Q1: (false) " + s1 + (q1_permutation(s1,s2) ? " is " : " isn't ") 
+                + "a permutation of " + s2);
+       
+       s1 = "a";
+       System.out.println("Q1: (true) " + s1 + (q1_permutation(s1,s2) ? " is " : " isn't ") 
+                + "a permutation of " + s2);
+       
+       s2 = "aa";
+       System.out.println("Q1: (false) " + s1 + (q1_permutation(s1,s2) ? " is " : " isn't ") 
+                + "a permutation of " + s2);
+       
+       s1 = "aaa";
+       System.out.println("Q1: (false) " + s1 + (q1_permutation(s1,s2) ? " is " : " isn't ") 
+                + "a permutation of " + s2);
+       
+       s2 = "";
+       System.out.println("Q1: (false) " + s1 + (q1_permutation(s1,s2) ? " is " : " isn't ") 
+                + "a permutation of " + s2);
+       
+       s1=" ";
+       System.out.println("Q1: (false) " + s1 + (q1_permutation(s1,s2) ? " is " : " isn't ") 
+                + "a permutation of " + s2);
+       
+       s1="abc";
+       s2="bcd";
+       System.out.println("Q1: (false) " + s1 + (q1_permutation(s1,s2) ? " is " : " isn't ") 
+                + "a permutation of " + s2);
+             
+        s1="Listen";
+        s2="Silent";
+        System.out.println("Q1: (true) " + s1 + (q1_permutation(s1,s2) ? " is " : " isn't ") 
+                + "a permutation of " + s2);
+       
+       
+        s1="Triangle";
+        s2="Integral";
+        System.out.println("Q1: (true) " + s1 + (q1_permutation(s1,s2) ? " is " : " isn't ") 
+                + "a permutation of " + s2);
+       
+        s1="Apple";
+        s2="Pabble";
+        System.out.println("Q1: (false) " + s1 + (q1_permutation(s1,s2) ? " is " : " isn't ") 
+                + "a permutation of " + s2);
+    }

```

Comment:
These are not really automated tests. They feed in parameters, but they don't actually _check_ the result, just print it.

If you include an expected result and check that, then the tests are automated, and you can modify your code and run the tests to check it still works much more easily.